feat(dev): added git hook restage for adding files

WarpWing opened the pull request:
**Pull Request Description**

Pursuant to #32, patched back changes from #412 

Closes #32 
---

**Licensing Certification**

FarmData2 is a [Free Cultural Work](https://freedomdefined.org/Definition) and all accepted contributions are licensed as described in the LICENSE.md file. This requires that the contributor holds the rights to do so. By submitting this pull request **I certify that I satisfy the terms of the [Developer Certificate of Origin](https://developercertificate.org/)** for its contents.

WarpWing committed:
feat(dev): patch back changes to update pre-commit githook to re-add staged files pursuant to #32

Signed-off-by: Ty Chermsirivatana <chermsit@dickinson.edu>

WarpWing committed:
fix(dev): revamped logic to detect changes instead of checking file "sameness" for clarity

Signed-off-by: Ty Chermsirivatana <chermsit@dickinson.edu>

braughtg commented on .githooks/pre-commit:
```diff
@@ -22,8 +21,40 @@ if [ "$BRANCH" = "development" ]; then
 # For all other branches, run the pre-commit checks using lint-staged.
 if npm run check:staged; then
+  # Check if any staged files were modified during the pre-commit checks
+  if check_modified_staged_files; then
```
This if should come before and be independent from the `if npm run check:staged" conditional. 

The sequence of events we want to address is:
- The user stages some changed files.
- The user tries to commit the staged changes
- The`npm run check:staged` fails and the hook exists
- The user makes changes to some of the files that were staged to address the issue(s).
- The user tries to commit again (but forgot to `git add` the changes).
  - This is when the check should find the files that were previously staged but which were not re staged after the new changes.

WarpWing committed:
fix(dev): moved git staged file check files before running npm check stage.

Signed-off-by: Ty Chermsirivatana <chermsit@dickinson.edu>

WarpWing commented:
@braughtg As per your changes requested, I've moved the git stage file check above the `npm run check:staged` which now flows as 

```
Check for staged files --> ask user to stage if found --> only then proceed with pre-commit checks --> exit as appropriate
```

I've also moved the function definition to the top of the script before the branch checks.

braughtg committed:
patch script to accept input and restage files

braughtg commented:
@WarpWing I pushed some changes to this PR.  Two key points:
- The `read` command was not working in the `git-hook`
- All files were being staged, not just the ones that were previously staged.

I also revised the wording of things a bit and added some output that lists the files.

To give it a test:
- pull your branch
- create a new branch from it
- modify the README.md file with a bad change (e.g. a spelling error)
- try to commit - this will fail.
- fix the error README.md file but do not restage it.
- try to commit again.

 Let me know if you see anything that's an issue.

WarpWing commented:
Just tested a clean run, this looks good to me!

WarpWing committed:
Merge branch 'development' into patch-8

WarpWing commented:
This checks out on my end!
